sklearn/ensemble/_hist_gradient_boosting/_binning.pyx

```diff
@@ -0,0 +1,58 @@
+# cython: cdivision=True
+# cython: boundscheck=False
+# cython: wraparound=False
+# cython: nonecheck=False
+# cython: language_level=3
+
+# Author: Nicolas Hug
+
+cimport cython
+
+import numpy as np
+cimport numpy as np
+from cython.parallel import prange
+
+from .types cimport X_DTYPE_C, X_BINNED_DTYPE_C
+
+cpdef _map_to_bins(const X_DTYPE_C [:, :] data, list binning_thresholds,
+                   X_BINNED_DTYPE_C [::1, :] binned):
+    """Bin numerical values to discrete integer-coded levels.
+
+    Parameters
+    ----------
+    data : array-like, shape=(n_samples, n_features)
```

Comment:
Is it really array-like, meaning that you can pass a list, or just a typed ndarray.